src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -737,32 +739,55 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref Compo
                     case TypeKind.Struct:
                     case TypeKind.Interface:
                     case TypeKind.Delegate:
-                        var typeArguments = ((NamedTypeSymbol)current).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
-                        if (typeArguments.IsEmpty)
+
+                        if (current.IsAnonymousType && current.TypeKind == TypeKind.Class)
```

Comment:
>&& current.TypeKind == TypeKind.Class [](http://example.com/codeflow?start=52&length=37)

I think we shouldn't have this condition.

Reply:
Actually I have no idea how to inspect synthesized delegates.

Reply:
> Actually I have no idea how to inspect synthesized delegates.

I think they share the same data structures and the implementation doesn't really have to change.

Reply:
We do want to add a test for a scenario with a delegate. Similar to the one with a regular anonymous type.

Reply:
For anonymous classes we need the types of their properties, for delegates we need their types of the method signature. How to get those? Could you please share some examples?

Reply:
> For anonymous classes we need the types of their properties, for delegates we need their types of the method signature. How to get those? Could you please share some examples?

I already responded in a previous message on this thread. Here is the quote: "I think they share the same data structures and the implementation doesn't really have to change."

Reply:
> We do want to add a test for a scenario with a delegate. Similar to the one with a regular anonymous type.

This PR contains several in the emit tests, no combined, i.e. anonymous class within anonymous delegate. Do we have existing tests for this helper? If so I'm happy to add several for it.

Reply:
>> Actually I have no idea how to inspect synthesized delegates.
>
> I think they share the same data structures

Thanks, I figured it out from example here: https://github.com/dotnet/roslyn/blob/05b2c023ffebacb67c776d6b7be4c77db3a4dd18/src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/PublicSymbols/AnonymousType.DelegatePublicSymbol.cs#L52-L65